RDoc-1591 documented rules for naming of CMPXCHG keys, indexes, and databases.

morhilai opened the pull request.

ayende commented:
@karmeli87 Please review this

arekpalinski commented:
@morhilai When copying files from earlier version please do it in a separate commit. Then the actual changes should be introduced in next commit. This way a reviewer can look _only_ at the relevant changes. Right now it's hard to see what exactly has been changed here because everything is seen as a new file.

karmeli87 commented:
looks good

arekpalinski commented:
Please check travis CI build error:

```
Unhandled Exception: System.InvalidOperationException: DiscussionId is missing in C:\Users\travis\build\ravendb\docs\Documentation\4.1\Raven.Documentation.Pages\indexes\.docs.json for item: creating-and-deploying. Please generate the documentation again and make sure to commit the .docs.json file changes.```

You can ask @topalach if needed

morhilai commented:
Fixed the build error. 
Also added the same text about index naming rules to `Indexes: Creating and Deploying` at the end of the static index section
